cpp/modules/luma.gl/garrow/src/key-value-metadata.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2020 Unfolded Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "./key-value-metadata.h"  // NOLINT(build/include)
+
+using namespace lumagl::garrow;
+
+KeyValueMetadata::KeyValueMetadata() {}
```

Comment:
Any reason we are not using the arrow class (that I assume this is being copied from)? I believe we still are dependent on arrow API?

Reply:
We're not dependent on `arrow` API at all thus far, other than `arrow-utils` transforms and the temporary convenience `Array` data sources. I don't see a need to mix the two unless we can leverage a large chunk of functionality, but in this case it was simple enough to port